jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/kubernetes/KubernetesUndeployService.java

```diff
@@ -82,19 +82,21 @@ private void undeployResources(String namespace, List<HasMetadata> entities) {
 
   protected Consumer<HasMetadata> resourceDeleter(String namespace) {
     return resource -> {
-      logger.info("Deleting resource %s %s/%s", KubernetesHelper.getKind(resource), namespace, KubernetesHelper.getName(resource));
+      String undeployNamespace = resolveEffectiveNamespace(resource, namespace);
```

Comment:
Why isn't the namespace resolved after line 68 and passed on? (Just like we're doing now)

Reply:
This is done to handle the case of multiple namespaces. In case resources have different namespaces in their metadata, it's given more precedence over standard namespace.